app/assets/javascripts/users.js

```diff
@@ -0,0 +1,75 @@
+$(document).ready(function() {
+	name_state = false;
+	email_state = false;
+	password_state = false;
+
+	$("#show_password").change(function() {
+		if ($(this).is(':checked'))
+		{
+			$(".pword").attr("type","text");
+		}
+		else
+		{
+			$(".pword").attr("type","password");
+		};
+	});
+
+	$("#user_first_name, #user_last_name").on("focusout keyup", function() {
+		if ( $(this).val() === "")
+		{
+			$(this).addClass("invalid");
+			$(this).prev(".field-error").text("Field is required");
+			name_state = false;
+		}
+		else{
+			$(this).removeClass("invalid");
+			$(this).prev(".field-error").text("");
+			name_state = true;
+		}
+			changeButton();
+	});
+
+	$("#user_email").on("focusout keyup", function() {
+		var pattern = /^([a-z\d_\.\-])+\@(([a-z\d\-])+\.)+([a-z]{2,4})+$/i;
+		var email = $(this).val();
+		if (!pattern.test(email)) {
+			$(this).addClass("invalid");
+			$(this).prev(".field-error").text("Email is invalid");
+			email_state = false;
+		}
+		else{
+			$(this).removeClass("invalid");
+			$(this).prev(".field-error").text("");
+			email_state = true;
+		}
+		changeButton();
+	});
+
+	$("#user_password").on("focusout keyup", function() {
+		var password = $(this).val();
+		if (password.length < 6){
+			$(this).addClass("invalid");
+			$(this).prev(".field-error").text("Password " +
+				"must be more than 6 characters");
+			password_state = false;
+		}
+
+		else{
+			$(this).removeClass("invalid");
+			$(this).prev(".field-error").text("");
+			password_state = true;
+		}
+		changeButton();
+	});
+
+	function changeButton () {
+		if (password_state && email_state && name_state)
```

Comment:
Identifier 'password_state' is not in camel case.<br>Identifier 'email_state' is not in camel case.<br>Identifier 'name_state' is not in camel case.<br>'password_state' is not defined.<br>'email_state' is not defined.<br>'name_state' is not defined.